### PR TITLE
BK-709 Add Indices CRUD methods in PHP library

### PR DESCRIPTION
**Pull request by GLagunilla**

### CODE REVIEW

**Comment by GLagunilla** [comment]
:warning: Avoid review files from doc folder. It is just documentation. :warning:

File: src/Management/Model/DataSource.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /**
+     * @param array $data
+     * @return DataSource
+     */
+    public static function createFromArray(array $data)
+    {
+        /** @var class-string<DataSourceOption> $fqcn */
+        $fqcn = self::FQCN_OPTIONS[$data['type']];
```
Comment (mrroot5) [inline src/Management/Model/DataSource.php:50]: Could be possible that `$data['type']` does not exists on `FQCN_OPTIONS` and this code fails?

**Review by mrroot5 — APPROVED** [review]

File: src/Management/Model/DataSource.php (hunk 1 above)
Comment (GLagunilla) [reply src/Management/Model/DataSource.php:50]: It's not possibl because there are 5 different types, included in FQCN_OPTIONS constant.

Comment (mrroot5) [reply src/Management/Model/DataSource.php:50]: Ok, thanks :-)

**Review by sonic182 — APPROVED** [review]
I only see the ManagementClient test file big, but appart from that it seems ok

**Comment by GLagunilla** [comment]
> I only see the ManagementClient test file big, but appart from that it seems ok

Yes, it's too big :thinking: 
In the next PR I will think about split the file.

**Pull request merged**